planner/mapper/mapper.go

```diff
@@ -743,7 +806,14 @@ func resolveInnerFilterDependencies(
 
 		if keyIndex >= len(mapping.ChildMappings) {
 			// If the key index is outside the bounds of the child mapping array, then
-			// this is not a relation/join and we can continue (no child props to process)
+			// this is not a relation/join and we can add it to the fields and
+			// continue (no child props to process)
+
+			newFields = append(existingFields, &Field{
```

Comment:
question: I cannot tell why this has been added - can you please explain?

Reply:
The main theme of most (all?) of the `mapper` changes is that it originally didn't explicitly state a lot of the dependent fields that are indirectly required through filters, orders, aggregates, groups, etc. Since before, the `fetcher` just always *all* fields, so we didn't need to be picky.

This particular change will add a related type field defined on a parent filter to the child selection set.
ie:
```graphql
query {
  User(filter: {friend: {age: _eq: 10}} {
    name
  }
}
```

This query would generate a virtual sub select as we dont include the `friend` sub type in the selection set, however, before it basically would be an empty selections set on the `friend` type, because again, the fetcher would automatically return all fields. 

With the new changes to the fetcher, we need to include the fields on the virtual sub select that we care about.

Reply:
Ah of course - and the 'new' leaf fields are added to the  `ChildMappings` - thanks for the explanation. I wonder if it is worth rewording the code-comment a little more to make this a bit more clear?

Reply:
> Ah of course - and the 'new' leaf fields are added to the ChildMappings - thanks for the explanation. I wonder if it is worth rewording the code-comment a little more to make this a bit more clear?

Can update the comment. Small note, the new leaf fields are added to the `mapper.Select` type, not the ChildMapping.

Reply:
cheers :)